project2-beat-that/index.html

```diff
@@ -60,42 +60,79 @@
       <br />
       <!-- The <button> tag represents.. you guessed it! a button. -->
-      <button id="submit-button">Submit</button>
+      <button id="roll-button">Roll</button>
+      <button id="restart-button">Restart</button>
       <p>Output:</p>
       <div class="output-div" id="output-div"></div>
+      <p>Score:</p>
+      <div class="output-div" id="outputscore-div"></div>
     </div>
+    <script>
+      var output = document.querySelector("#output-div");
+      output.innerHTML = "Welcome to Beat That! Player 1, roll your dice!";
+      var outputscore = document.querySelector("#outputscore-div");
+      outputscore.innerHTML =
+        "Leaderboard" + "<br>" + "Player 1: 0" + "<br>" + "Player 2: 0";
+    </script>
     <!-- Everything below this is incorporating JavaScript programming logic into the webpage. -->
     <!-- The script tag encloses JavaScript syntax for the browser to interpret programtically -->
     <!-- The following line imports all code written in the script.js file -->
     <script src="script.js"></script>
     <!-- Define button click functionality -->
     <script>
       // Declare and define a variable that represents the Submit Button
-      var button = document.querySelector("#submit-button");
+      var rollbutton = document.querySelector("#roll-button");
       // When the submit button is clicked, access the text entered by the user in the input field
       // And pass it as an input parameter to the main function as defined in script.js
-      button.addEventListener("click", function () {
+      rollbutton.addEventListener("click", function () {
         // Set result to input value
-        var input = document.querySelector("#input-field");
+        // var input = document.querySelector("#input-field");
         // Store the output of main() in a new variable
-        var result = main(input.value);
+        var roundresult = main();
+        var scoreresult = main("score");
 
         // Display result in output element
         var output = document.querySelector("#output-div");
-        output.innerHTML = result;
+        output.innerHTML = roundresult;
+
+        var outputscore = document.querySelector("#outputscore-div");
+        outputscore.innerHTML = scoreresult;
+
+        // Reset input value
+        // input.value = "";
+      });
+      var restartbutton = document.querySelector("#restart-button");
+      restartbutton.addEventListener("click", function () {
+        // Set result to restart
+        // var input = document.querySelector("#input-field");
+        // Store the output of main() in a new variable
+        var roundresult = main("restart");
+        var scoreresult = main("resetscore");
+
+        // Display result in output element
+        var output = document.querySelector("#output-div");
+        output.innerHTML = roundresult;
+
+        var outputscore = document.querySelector("#outputscore-div");
+        outputscore.innerHTML = scoreresult;
```

Comment:
Good that you have tried adding a new button! You don't have to use the main function , you can create a separate function to handle your restart and resetscore . 

This will abstract out your game logic and your game flow so that it will be easier to read in the code.